waiter/src/waiter/cookie_support.clj

```diff
@@ -19,35 +19,19 @@
             [taoensso.nippy :as nippy]
             [waiter.utils :as utils])
   (:import clojure.lang.ExceptionInfo
+           java.net.HttpCookie
            java.nio.charset.StandardCharsets
            org.eclipse.jetty.util.UrlEncoded))
 
-(defn url-decode
```

Comment:
I'll do some regression testing after incorporating the other feedback and post the results